### DISCUSSION TITLE
DietPi-Software | Amiberry 4.1.6

### DISCUSSION THREAD

**Discussion by wifi75** [MichaIng/DietPi#5083] [Software Request] [answered]
### Creating a feature request
hello when it is available amiberry 4.1.6?
#### Is your feature request related to a problem? Please describe:

- ...

#### Describe the solution you'd like:

- ...

#### Describe alternatives you've considered:

- ...

#### Additional context

- ...

**Comment by Joulinar** [comment]
We need to pre-compile the binaries. Pls be patient.

**Comment by MichaIng** [comment]
@Joulinar 
Let's please keep such update requests on issues. It is just a question when I find time to build and test and upload the new version, the voting system is not required here, the issue more a reminder and to ask others to test a build.

Hmm, I just see that the original issue cannot be opened anymore, only a new issue created based on this discussion but with original post authors lost 🤔.

**Comment by Joulinar** [comment]
@MichaIng 
would it be possible to create a 64bit version as well? At least it seems to be possible https://github.com/midwan/amiberry/wiki/Available-Platforms#raspberry-pi-models-64-bit-sdl2-with-dispmanx

**Reply by MichaIng** [reply]
Yes generally it is. Means I need to spin an RPi 64-bit container on the Odroid N2 😄. Will do that in a few days when I'm back home, together with the general Amiberry upgrade our end.

**Reply by Joulinar** [reply]
If you give me the instructions, I could do it on a RPi4B

**Reply by MichaIng** [reply]
I'll upload the build screen as well then, probably directly with an Amiberry DEB package as result. But it needs to be adjusted to assure it supports RPi 64-bit.

**Comment by actraiser30** [comment]
Hello there a quick question

what version of Amibbery is dietpi installing by default?

The dmx or the sdl2 version?

thank you

**Reply by MichaIng** [reply]
For RPi it ships with the DispmanX version.

**Reply by actraiser30** [reply]
thanks when you are going to update the amiberry version to 5.1 ?

thank you

**Reply by MichaIng** [reply]
@actraiser30 
On which SBC you want to run Amiberry, and if RPi: 32-bit or 64-bit image?

**Comment by MichaIng** [comment]
It's currently not the newest version. I'll do a build later for you to test. There have been quite some updates and I can only test a limited number of builds.

**Comment by MichaIng** [comment, marked as answer]
PR up, I'll also build the packages now for all platforms and distro versions: https://github.com/MichaIng/DietPi/pull/5481

**Comment by actraiser30** [comment]
Rpi4 32bit

Στις Σάβ, 14 Μαΐ 2022, 20:37 ο χρήστης MichaIng ***@***.***>
έγραψε:

**Reply by MichaIng** [reply]
Could you test this:
```sh
cd /tmp
curl -O https://dietpi.com/downloads/binaries/bullseye/amiberry_rpi4.deb
apt install ./amiberry_rpi4.deb
rm -Rf /mnt/dietpi_userdata/amiberry/lib/libSDL2*.so{,.0.*} # obsolete library files from previous Amiberry archive
```

**Reply by actraiser30** [reply]
thanks again sorry for the delay im working late, if i delete the lib/sdl2 libraries will amiberry still be working?

**Reply by MichaIng** [reply]
The removed files are obsolete, probably even conflicting. The old archive contained one library file + 2 symlink for each library, since different software/build stacks may look for a different naming/versioning. However, for Amiberry we know exactly which file names it is looking for, hence the new package contains only this one exact file 🙂.

**Reply by actraiser30** [reply]
i tested it, it works,version 5.2 released yesterday can you put that in the next update?

Thank you

**Reply by MichaIng** [reply]
I already saw it. Will implement Amiberry builds with GitHub actions to be done with this release already.

**Reply by wifi75** [reply]
hello how do you download the latest version of amiberry with sietpi?

**Reply by Joulinar** [reply]
Hi, pls try to stick to English language. Thx

On next DietPi release 8.5, you should be able to install/reinstall Amiberry using `dietpi-software`.

**Reply by wifi75** [reply]
hello how do you download the latest version of amiberry with sietpi?

**Reply by MichaIng** [reply]
You can either apply to the [beta](https://github.com/MichaIng/DietPi/issues/5511) or wait for the release this Saturday. Then reinstall Amiberry:
```sh
dietpi-software reinstall 108
```

**Reply by wifi75** [reply]
I downloaded and installed Amiberry ARMv7, but 8.5 is not present, in fact after having given the command dietpi-software reinstall 108 it always installs 4.1.5.
what am I doing wrong?

**Reply by Joulinar** [reply]
You are not using DietPi Beta version.

**Reply by wifi75** [reply]
ok I have to follow these steps
Beta on a fresh image:
Write the DietPi image to SD card.
Open the file /boot/dietpi.txt on the 1st partition.
Change DEV_GITBRANCH = master to DEV_GITBRANCH = beta (located near the bottom of the file).
Save the file, eject media and power on.

**Reply by Joulinar** [reply]
You can apply the beta version to existing installation as well.


```
G_CONFIG_INJECT 'DEV_GITBRANCH=' 'DEV_GITBRANCH=beta' /boot/dietpi.txt
```

**Reply by wifi75** [reply]
forgive me, but when I am in the terminal to return to amiberry what command should I execute?

**Reply by wifi75** [reply]
I tryed to send ./amiberry on /mnt/dietpi_userdata/amiberry but them show this error:

./amiberry: error while loading shared libraries: libSDL2-2.0.so.0: cannot open shared object file: No such file or directory

**Reply by Joulinar** [reply]
to start `Amiberry` you need to run following.

```
systemctl start amiberry
```

Anyway, pls open an own/new issue if you have challenges running `Amiberry`

**Reply by wifi75** [reply]
why install amiberry 5.1 and not 5.2?
5.2 is the last published

**Reply by MichaIng** [reply]
Because v5.2 was released 4 days ago while our packages were built 10 days ago 😉. I'll update them until the DietPi release.

**Reply by MichaIng** [reply]
All Amiberry v5.2 builds have finished, e.g. for Bullseye here: https://dietpi.com/downloads/binaries/bullseye/testing/
E.g. for RPi 4 32-bit (ARMv6 or ARMv7) image:
```sh
cd /tmp
curl -O https://dietpi.com/downloads/binaries/bullseye/testing/amiberry_rpi4.deb
dpkg -i amiberry_rpi4.deb
rm amiberry_rpi4.deb
```
Would be great if you could test it, I don't have all RPi models 🙂. I'll move them to stable location then, so it can be updated via regular reinstall then as well:
```sh
dietpi-software reinstall 108
```
